pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -846,61 +847,150 @@ func (m *kubeGenericRuntimeManager) purgeInitContainers(ctx context.Context, pod
 	}
 }
 
-// findNextInitContainerToRun returns the status of the last failed container, the
-// index of next init container to start, or done if there are no further init containers.
-// Status is only returned if an init container is failed, in which case next will
-// point to the current container.
-func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus) (status *kubecontainer.Status, next *v1.Container, done bool) {
+// computeInitContainerActions computes the actions need to take for init
+// conainers and returns the flag indicating whether the pod has been initialized.
+//
+// The actions include:
+// 1. The first init container that has not completed successfully will be
+// started.
+// 2. All sidecar containers that have started but are not running will be
+// restarted.
+// 3. Kill sidecar containers that have failed the startup probe.
```

Comment:
```suggestion
// computeInitContainerActions sets the actions on the podActions that need to be taken
// to for the init containers. This includes actions to initialize the init containers
// and actions to keep sidecar containers running. computeInitContainerActions
// returns true if pod has been initialized.
//
// The actions include:
// 1. The first init container that has not completed successfully will be
// started.
// 2. All sidecar containers that have started but are not running will be
// restarted.
// 3. Kill sidecar containers that have failed the startup probe.
```

?  Just a suggestion based on my reading of the implementation.

Reply:
updated it, thanks.